main/lsp/ErrorReporter.h

```diff
@@ -0,0 +1,35 @@
+#ifndef RUBY_TYPER_LSP_ERRORREPORTER_H
+#define RUBY_TYPER_LSP_ERRORREPORTER_H
+
+#include "core/core.h"
+#include "main/lsp/LSPConfiguration.h"
```

Comment:
(Nit) You can remove this include by forward declaring `LSPConfiguration` (it's only referred to via shared_ptr).